src/main.rs

```diff
@@ -1480,23 +1491,34 @@ fn get_setup() -> Setup {
             })
             .unwrap_or_default();
 
+        let volume_steps = opt_str(VOLUME_STEPS)
+            .map(|steps| match steps.parse::<u16>() {
+                Ok(value) => value,
+                _ => {
+                    let default_value = &connect_default_config.volume_steps.to_string();
+
+                    invalid_error_msg(
+                        VOLUME_STEPS,
+                        VOLUME_STEPS_SHORT,
+                        &steps,
+                        "a positive whole number <= 65535",
```

Comment:
As the volume steps seem to represent not the actual volume adjustment, but rather the amount of steps in between the volume range, might it be a better idea to change the underlying config type value to `u8` and by that restrict the maximum to `255` steps? Just noticed that `65535` don't make much sense in terms of steps in between the volume range^^;

But what are your thoughts on that?

Reply:
Looking at the API it seems the maximum number of steps supported [is in fact 65535](https://developer.spotify.com/documentation/commercial-hardware/implementation/reference/latest#spsetvolumesteps), given we don't know how individuals might want to use this player (i.e. connected to a PA I could imagine wanting some pretty fine grained control) it seems to me to make sense to support the full range and just change the default as I have to be more reasonable for an out of the box desktop user.

Reply:
Hmm, fair. Even tho I find it hard to believe someone would want more then 255 volume steps.

Reply:
Indeed, 64 steps is the official client behavior and a reasonable range for a stepped volume control. `65.536 / 64 = 1.024` so that may be where the confusion was coming from.